build.gradle

```diff
@@ -211,6 +211,8 @@ project ('spring-kafka-test') {
 
 		compile ("org.hamcrest:hamcrest-all:$hamcrestVersion", optional)
 		compile ("org.assertj:assertj-core:$assertjVersion", optional)
+
+        testCompile project(':spring-kafka')
```

Comment:
No, your test definitely should be in the `spring-kafka` module, in the `test` directory. And I think the `org.springframework.kafka.kstream` package should be good candidate for the place.